packages/devtools-reps/src/object-inspector/tests/object-inspector.js

```diff
@@ -741,4 +743,76 @@ describe("getChildren", () => {
     expect(names).toEqual(["<get>", "<set>"]);
     expect(paths).toEqual(["rootpath/get", "rootpath/set"]);
   });
+
+  it("uses the expected actor to get properties", () => {
+    const stub = performanceStubs.get("performance");
+    const getObjectProperties = jest.fn();
+
+    // Test that the function gets the actor from the value.
+    getChildren({
+      actors: {},
+      item: createNode("root", "rootpath", stub.ownProperties.timing),
+      getObjectProperties
+    });
+    expect(getObjectProperties.mock.calls[0][0]).toBe("server2.conn4.child1/obj44");
+
+    // Test that the function gets the actor from the getterValue.
+    getChildren({
+      actors: {},
+      item: createNode("root", "rootpath", stub.safeGetterValues.timing),
+      getObjectProperties
+    });
+    expect(getObjectProperties.mock.calls[1][0]).toBe("server2.conn4.child1/obj44");
+  });
+
+  it("safeGetterValues", () => {
+    const stub = performanceStubs.get("timing");
+    const nodes = getChildren({
+      actors: {},
+      item: createNode("root", "rootpath", {
+        value: {
+          actor: "rootactor",
+          type: "object"
+        }
+      }),
+      getObjectProperties: actor => {
+        if (actor === "rootactor") {
+          return stub;
+        }
+        return null;
+      }
+    });
+
+    const nodeEntries = nodes.map(n => [n.name, getValue(n)]);
+    const nodePaths = nodes.map(n => n.path);
+
+    const childrenEntries = [
+      ["connectEnd", 1500967716401],
+      ["connectStart", 1500967716401],
+      ["domComplete", 1500967716719],
+      ["domContentLoadedEventEnd", 1500967716715],
+      ["domContentLoadedEventStart", 1500967716696],
+      ["domInteractive", 1500967716552],
+      ["domLoading", 1500967716426],
+      ["domainLookupEnd", 1500967716401],
+      ["domainLookupStart", 1500967716401],
+      ["fetchStart", 1500967716401],
+      ["loadEventEnd", 1500967716720],
+      ["loadEventStart", 1500967716719],
+      ["navigationStart", 1500967716401],
+      ["redirectEnd", 0],
+      ["redirectStart", 0],
+      ["requestStart", 1500967716401],
+      ["responseEnd", 1500967716401],
+      ["responseStart", 1500967716401],
+      ["secureConnectionStart", 1500967716401],
+      ["unloadEventEnd", 0],
+      ["unloadEventStart", 0],
+      ["__proto__", stub.prototype]
```

Comment:
this is a good example where the snapshot approach could be easier to maintain. The next time you generate the stubs, you'll get different data, but it is not relevant for the test